mrf/js/56-data-fetching/mian.js

```diff
@@ -0,0 +1,102 @@
+// Define the API URL
+const apiUrl = "https://jsonplaceholder.typicode.com/";
+
+async function getAuthorPost(userID) {
+  const response = await fetch(apiUrl + `users/${userID}`);
+  const final = await response.json();
+
+  return final["name"];
+}
+
+async function getPost(id) {
+  const response = await fetch(apiUrl + `posts/${id}`);
+  const final = await response.json();
+
+  const author = await getAuthorPost(final["userId"]);
+  final["name"] = author;
+
+  return final;
+}
+
+function addCard(titleIn, textIn, authorIn, ID) {
+  const master = document.createElement("div");
+  master.setAttribute("class", "card");
+  master.setAttribute("id", ID);
+
+  const title = document.createElement("p");
+  title.textContent = titleIn;
+
+  const text = document.createElement("p");
+  text.textContent = textIn;
+
+  const author = document.createElement("p");
+  author.textContent = authorIn;
+
+  const deleteButton = document.createElement("button");
+  deleteButton.setAttribute("class", "delete");
+  deleteButton.addEventListener("click", () => {
+    //   deleteCard(event.target.parentNode.id);
+    deleteCard(ID);
+  });
+  deleteButton.textContent = "Delete";
+
+  const updateButton = document.createElement("button");
+  updateButton.setAttribute("class", "update");
+  updateButton.addEventListener("click", () => {
+    //   updateCard(event.target.parentNode.id);
+    updateCard(ID);
+  });
+  updateButton.textContent = "Update";
+
+  master.appendChild(title);
+  master.appendChild(text);
+  master.appendChild(author);
+  master.appendChild(deleteButton);
+  master.appendChild(updateButton);
+
+  return master;
+}
+
+function deleteCard(masterID) {
+  if (window.confirm("Do you really want to Delete?")) {
+    const masterDIV = document.getElementById(masterID);
+    document.body.removeChild(masterDIV);
+
+    fetch(`https://jsonplaceholder.typicode.com/posts/${masterID}`, {
```

Comment:
This may fail. You should remove the div after fetch is successful.

Reply:
got it 🌾